toolchain/sem_ir/name_scope.h

```diff
@@ -67,33 +67,53 @@ class ScopeLookupResult {
   // True when lookup was successful or resulted with an error. False for
   // poisoned or not found.
   auto is_found() const -> bool {
-    return !is_poisoned() && inst_id_.has_value();
+    return !is_poisoned() && target_inst_id_.has_value();
   }
 
   // The `InstId` of the result of the lookup. Must only be called when lookup
   // was successful e.g. `is_found()` returns true. Always returns an existing
   // `InstId`.
   auto target_inst_id() const -> InstId {
     CARBON_CHECK(is_found());
-    return inst_id_;
+    return target_inst_id_;
+  }
+
+  // The `LocId` where the name poisoning was triggered. Must only be called
+  // when lookup resulted with a poisoned name e.g. `is_poisoned()` returns
+  // true. Always returns an existing `InstId`.
+  auto poisoning_loc_id() const -> LocId {
+    CARBON_CHECK(is_poisoned());
+    return poisoning_loc_id_;
   }
 
   auto access_kind() const -> AccessKind { return access_kind_; }
 
   // Equality means either:
   // - Both are not poisoned and have the same `InstId` and `AccessKind`.
-  // - Both are poisoned.
-  friend auto operator==(const ScopeLookupResult&, const ScopeLookupResult&)
-      -> bool = default;
+  // - Both are poisoned and have the same `LocId`.
```

Comment:
Before, you were still verifying the access kind was the same, but couldn't this be a little simpler and just compare the access kind? i.e., it *should* always be the same -- given this is essentially for tests, maybe it's worth just verifying?

Reply:
Done.